debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java

```diff
@@ -225,12 +225,21 @@ protected String idObjToJson(Object idObj) {
             if (idObj == null) {
                 return null;
             }
-            if (!(idObj instanceof Document)) {
+            if (idObj instanceof Document) {
+                Object idValue = ((Document) idObj).get(DBCollection.ID_FIELD_NAME);
+                /*
+                 * Special treatment for String as com.mongodb.util.StringSerializer
+                 * adds quotes around original string
+                 */
+                if(idValue instanceof String) {
+                    return (String) idValue;
+                } else {
+                    return jsonSerializer.serialize(idValue);
+                }
+
+            } else {
                 return jsonSerializer.serialize(idObj);
```

Comment:
Looking at the code it can be simplified, just make a special `return` for the String branch and a common `return` for the rest.

Reply:
Not a fan of default returns but it is your house -> your rules